### PR TITLE
Support Color Temperature

### PR DESCRIPTION
**Pull request by AlCalzone**
This PR is intended to add support for the [ColorTemperatureController](https://developer.amazon.com/de/docs/device-apis/alexa-colorcontroller.html#setcolor) interface. Its a work in progress as I do need some help getting this to the finish line.

In the first commit I added basic support for setting a color temperature. Increment/Decrement comes later when I know how to proceed.

### CODE REVIEW

File: lib/alexaSmartHomeV2.js [hunk 1]
```diff
@@ -276,20 +278,27 @@ function AlexaSH2(adapter) {
...
                 if (type === 'number') {
-                    if (states[id].common.unit === 'C' || states[id].common.unit === 'C°' || states[id].common.unit === '°C' ||
-                        states[id].common.unit === 'F' || states[id].common.unit === 'F°' || states[id].common.unit === '°F' ||
-                        states[id].common.unit === 'K' || states[id].common.unit === 'K°' || states[id].common.unit === '°K') {
+                    if (/^level\.color\./.test(role)) {
+                        // test for color first, since color temperature may be given in Kelvin (K)
+                        if (role === 'level.color.hue' || role === 'level.color.rgb') {
+                            actions = ['setColor', /*'incrementColorTemperature', 'decrementColorTemperature', 'setColorTemperature', */'turnOn', 'turnOff'];
+                        } else if (role === 'level.color.temperature') {
+                            actions = ['setColorTemperature', /*'incrementColorTemperature', 'decrementColorTemperature'*/];
```
Comment (AlCalzone) [inline lib/alexaSmartHomeV2.js]: Where are those actions defined? Does the server code also have to be updated to support those?

File: lib/alexaSmartHomeV2.js [hunk 2]
```diff
@@ -1447,6 +1522,24 @@ function AlexaSH2(adapter) {
...
+            case 'SetColorTemperatureRequest':
+                adapter.log.debug('ALEXA SetColorTemperature: ' + request.payload.appliance.applianceId + ' ' + request.payload.colorTemperatureInKelvin + '%');
+                for (i = 0; i < ids.length; i++) {
+                    controlColorTemperature(ids[i], request.payload.colorTemperatureInKelvin, ids.length === 1, function (err) {
+                        if (!--count) {
+                            request.header.name = 'SetColorTemperature';
+
+                            callback({
+                                header: request.header,
+                                payload: {}
```
Comment (AlCalzone) [inline lib/alexaSmartHomeV2.js]: The code for `SetColorRequests` responds with the state it achieved. What does that do? And how does the payload object for this have to look like?

File: lib/alexaSmartHomeV2.js [hunk 3]
```diff
@@ -105,10 +105,20 @@ function AlexaSH2(adapter) {
...
-                            text = obj.friendlyName + ' set to ' + value + ' <emphasis>percent</emphasis>';
+                            text = obj.friendlyName + ' <emphasis>set to</emphasis> ' + value + ' percent';
                         }
                         break;
 
+                    case 'K':
+                        if (lang === 'de') {
+                            text = obj.friendlyName + ' wird auf ' + value + ' Kelvin <emphasis>gesetzt</emphasis>';
+                        } else if (lang === 'ru') {
+                            text = 'Состояние ' + obj.friendlyName + ' <emphasis>установлено</emphasis> на ' + value;
```
Comment (AlCalzone) [inline lib/alexaSmartHomeV2.js]: This is the line from `percent` but I can't read russian :D

File: lib/alexaSmartHomeV2.js (hunk 1 above)
Comment (WolfspiritM) [reply lib/alexaSmartHomeV2.js]: You're working with the v2 skill api (which is depreacted and you actually can't create a skill with that version anymore) not the controller you're looking at (v3). The Actions are defined in the v2 Skill Documentation which in return is very hard to find right now...

These actions are defined here:
https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/archive/shv2-smart-home-skill-api-reference#discovery-messages
Scroll down to: discoveredAppliance.actions

File: lib/alexaSmartHomeV2.js (hunk 2 above)
Comment (WolfspiritM) [reply lib/alexaSmartHomeV2.js]: In V2 all "Set...Request" are returning a "Set...Confirmation". So you should change "SetColorTemperature" in the request.header.name to "SetColorTemperatureConfirmation".

See here: https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/archive/shv2-smart-home-skill-api-reference#setcolortemperatureconfirmation

File: lib/alexaSmartHomeV2.js (hunk 3 above)
Comment (WolfspiritM) [reply lib/alexaSmartHomeV2.js]: I'm a bit unsure what all the text messages should do anyway. The Alexa Smart Home Skill at least in v2 (I'm not sure about v3) can't return any text (maybe for Locks and Temperature) and Alexa always answers with "Ok" (except if it fails). As far as I can tell the only thing the writeResponse thing does is write that text to a state. But I have not yet figured out what that state is good for...

Comment (AlCalzone) [reply lib/alexaSmartHomeV2.js]: I think that's only for ioBroker if the user wants to evaluate what Alexa/ioBroker.cloud did, e.g. for visualization.

File: lib/alexaSmartHomeV2.js (hunk 1 above)
Comment (AlCalzone) [reply lib/alexaSmartHomeV2.js]: Thanks a lot!

**Comment by AlCalzone** [comment]
PR #40 is a more mature version of this PR and fixes some other issues too. I vote for merging PR #40 instead of this one and #28.

**Pull request closed without merging**